[Tools][JHBuild] Remove default outdated modules and default to minimal+gstreamer

clopez opened the pull request:
#### ab156c30e1d9f716f6886dd5515be7978670a7be
<pre>
[Tools][JHBuild] Remove default outdated modules and default to minimal+gstreamer
<a href="https://bugs.webkit.org/show_bug.cgi?id=302358">https://bugs.webkit.org/show_bug.cgi?id=302358</a>

Reviewed by Philippe Normand.

The default JHBuild modules are very outdated and unmaintained. It is better to
just delete them and default instead to the minimal+gstreamer ones that are in
better shape.

* Tools/gtk/jhbuild.modules:
* Tools/gtk/jhbuildrc:
* Tools/gtk/patches/at-spi2-atk-tests-Replace-common-variable-with-usual-external.patch: Removed.
* Tools/gtk/patches/at-spi2-atk-tests-fix-child_pid-reference.patch: Removed.
* Tools/gtk/patches/cairo-ft-Use-FT_Done_MM_Var-instead-of-free-when-available.patch: Removed.
* Tools/gtk/patches/fontconfig-C-11-requires-a-space-between-literal-and-identifier.patch: Removed.
* Tools/gtk/patches/icudata-stdlibs.patch: Removed.
* Tools/gtk/patches/libgcrypt-use-only-dev-urandom-for-testing.patch: Removed.
* Tools/gtk/patches/libgpg-error-prepare-for-gawk-5-0.patch: Removed.
* Tools/gtk/patches/libjxl-add-cmake-flag-provision-dependencies.patch: Removed.
* Tools/gtk/patches/libsecret-secret-methods-Don-t-unref-NULL-when-search-fails.patch: Removed.
* Tools/gtk/patches/mesa-gallium-build-fix-to-make-gcc-10-happy.patch: Removed.
* Tools/gtk/patches/mesa-gallivm-Fix-build-after-LLVM-commit-211259.patch: Removed.
* Tools/gtk/patches/xserver-Fix-building-with-fno-common.patch: Removed.
* Tools/gtk/patches/xserver-gl-version.patch: Removed.
* Tools/gtk/patches/xserver-remove-bogus-dependencies.patch: Removed.
* Tools/gtk/patches/xserver-search-for-DRI-drivers-at-LIBGL_DRIVERS_PATH-environ.patch: Removed.
* Tools/wpe/jhbuild.modules:
* Tools/wpe/patches/cairo-ft-Use-FT_Done_MM_Var-instead-of-free-when-available.patch: Removed.
* Tools/wpe/patches/fontconfig-C-11-requires-a-space-between-literal-and-identifier.patch: Removed.
* Tools/wpe/patches/gdate-suppress-string-format-literal-warning.patch: Removed.
* Tools/wpe/patches/icudata-stdlibs.patch: Removed.
* Tools/wpe/patches/libgcrypt-use-only-dev-urandom-for-testing.patch: Removed.
* Tools/wpe/patches/libgpg-error-prepare-for-gawk-5-0.patch: Removed.
* Tools/wpe/patches/libjxl-add-cmake-flag-provision-dependencies.patch: Removed.

Canonical link: <a href="https://commits.webkit.org/302991@main">https://commits.webkit.org/302991@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0a658e2278161ccc7509d698ae87fe4123a042a4

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/130710 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/3028 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/41665 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/138134 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/82355 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/132581 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/2998 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/2874 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/99580 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/67449 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/133657 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/2186 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/117032 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/80288 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/2108 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/35165 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/81387 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/110696 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/35666 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/140611 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/2771 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/2498 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/108085 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/2815 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/113374 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/108019 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/2131 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/31799 "Passed tests") | [❌ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/55768 "Hash 0a658e22 for PR 53775 does not build (failure)") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/20371 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/2841 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/2661 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/2862 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/2767 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

dpino reviewed (approved):
Looks good to me.

I think it's also possible to remove the following patches:

```
Tools/wpe/patches/icudata-stdlibs.patch
Tools/gtk/patches/icudata-stdlibs.patch
Tools/wpe/patches/libjxl-add-cmake-flag-provision-dependencies.patch
Tools/gtk/patches/libjxl-add-cmake-flag-provision-dependencies.patch
```

The JHBuild moduleset has its own copy of those patches:

```
Tools/jhbuild/patches/icudata-stdlibs.patch
Tools/jhbuild/patches/libjxl-add-cmake-flag-provision-dependencies.patch
```

clopez committed:
[Tools][JHBuild] Remove default outdated modules and default to minimal+gstreamer
https://bugs.webkit.org/show_bug.cgi?id=302358

Reviewed by Philippe Normand.

The default JHBuild modules are very outdated and unmaintained. It is better to
just delete them and default instead to the minimal+gstreamer ones that are in
better shape.

* Tools/gtk/jhbuild.modules:
* Tools/gtk/jhbuildrc:
* Tools/gtk/patches/at-spi2-atk-tests-Replace-common-variable-with-usual-external.patch: Removed.
* Tools/gtk/patches/at-spi2-atk-tests-fix-child_pid-reference.patch: Removed.
* Tools/gtk/patches/cairo-ft-Use-FT_Done_MM_Var-instead-of-free-when-available.patch: Removed.
* Tools/gtk/patches/fontconfig-C-11-requires-a-space-between-literal-and-identifier.patch: Removed.
* Tools/gtk/patches/icudata-stdlibs.patch: Removed.
* Tools/gtk/patches/libgcrypt-use-only-dev-urandom-for-testing.patch: Removed.
* Tools/gtk/patches/libgpg-error-prepare-for-gawk-5-0.patch: Removed.
* Tools/gtk/patches/libjxl-add-cmake-flag-provision-dependencies.patch: Removed.
* Tools/gtk/patches/libsecret-secret-methods-Don-t-unref-NULL-when-search-fails.patch: Removed.
* Tools/gtk/patches/mesa-gallium-build-fix-to-make-gcc-10-happy.patch: Removed.
* Tools/gtk/patches/mesa-gallivm-Fix-build-after-LLVM-commit-211259.patch: Removed.
* Tools/gtk/patches/xserver-Fix-building-with-fno-common.patch: Removed.
* Tools/gtk/patches/xserver-gl-version.patch: Removed.
* Tools/gtk/patches/xserver-remove-bogus-dependencies.patch: Removed.
* Tools/gtk/patches/xserver-search-for-DRI-drivers-at-LIBGL_DRIVERS_PATH-environ.patch: Removed.
* Tools/wpe/jhbuild.modules:
* Tools/wpe/patches/cairo-ft-Use-FT_Done_MM_Var-instead-of-free-when-available.patch: Removed.
* Tools/wpe/patches/fontconfig-C-11-requires-a-space-between-literal-and-identifier.patch: Removed.
* Tools/wpe/patches/gdate-suppress-string-format-literal-warning.patch: Removed.
* Tools/wpe/patches/icudata-stdlibs.patch: Removed.
* Tools/wpe/patches/libgcrypt-use-only-dev-urandom-for-testing.patch: Removed.
* Tools/wpe/patches/libgpg-error-prepare-for-gawk-5-0.patch: Removed.
* Tools/wpe/patches/libjxl-add-cmake-flag-provision-dependencies.patch: Removed.

Canonical link: https://commits.webkit.org/302991@main
